lib/beaker/host.rb

```diff
@@ -224,13 +225,29 @@ def is_x86_64?
     end
 
     def connection
-      @connection ||= SshConnection.connect( reachable_name,
+      # create new connection object if necessary
+      @connection ||= SshConnection.connect( { :ip => self['ip'], :vmhostname => self['vmhostname'], :hostname => @name },
                                              self['user'],
                                              self['ssh'], { :logger => @logger } )
+      # update connection information
+      if self['ip'] && (@connection.ip != self['ip'])
```

Comment:
Any reason not to just set `@connection.ip` to `self['ip']` if the latter has a value? Is there any reason to compare the two to determine if the same or not?